File: backend/models/Workout.js
const mongoose = require('mongoose');

const workoutSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true
  },
  startTime: {
    type: Date,
    required: true
  },
  endTime: {
    type: Date,
    required: true
  },
  distance: {
    type: Number,
    required: true
  },
  duration: {
    type: Number,  // in milliseconds
    required: true
  },
  steps: {
    type: Number,
    required: true
  },
  averageSpeed: {
    type: Number,
    required: true
  },
  calories: {
    type: Number,
    required: true
  },
  averageHeartRate: {
    type: Number,
    required: true
  },
  maxHeartRate: {
    type: Number,
    required: true
  },
  locations: [{
    latitude: Number,
    longitude: Number,
    timestamp: Date
  }]
}, {
  timestamps: true
});

module.exports = mongoose.model('Workout', workoutSchema); 

File: app/src/main/java/com/fittracker/models/Workout.kt
package com.fittracker.models

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Workout(
    @SerializedName("_id") val id: String? = null,
    val userId: String,
    val startTime: Date,
    val endTime: Date,
    val distance: Float,
    val duration: Long,
    val steps: Int,
    val averageSpeed: Float,
    val calories: Int,
    val averageHeartRate: Int,
    val maxHeartRate: Int,
    val locations: List<LocationPoint>,
    val createdAt: Date? = null,
    val updatedAt: Date? = null
)

data class LocationPoint(
    val latitude: Double,
    val longitude: Double,
    val timestamp: Date
) 